src/main/java/org/opensearch/knn/plugin/script/KNNScoringUtil.java

```diff
@@ -148,20 +140,28 @@ public static float cosineSimilarity(List<Number> queryVector, KNNVectorScriptDo
      */
     public static float cosinesimil(float[] queryVector, float[] inputVector) {
         requireEqualDimension(queryVector, inputVector);
-        float dotProduct = 0.0f;
-        float normQueryVector = 0.0f;
-        float normInputVector = 0.0f;
-        for (int i = 0; i < queryVector.length; i++) {
-            dotProduct += queryVector[i] * inputVector[i];
-            normQueryVector += queryVector[i] * queryVector[i];
-            normInputVector += inputVector[i] * inputVector[i];
+        int numZeroInInput = 0;
```

Comment:
I think these are unnecessary: https://github.com/apache/lucene/blob/main/lucene/core/src/java/org/apache/lucene/util/VectorUtil.java#L79. Can we just do this?

Reply:
That method would still return true for a zero vector right?

Reply:
I believe cosine will be infinite if one vector is finite

Reply:
As long as we validate that it's not zero vector in the above method, we should be able to remove the other check because of the assert finite